arcgis-ios-sdk-samples/Layers/Apply raster rendering rule/ApplyRasterRenderingRuleViewController.swift

```diff
@@ -0,0 +1,116 @@
+    @IBOutlet var chooseRenderingRuleBarButtonItem: UIBarButtonItem!
+    
+    // MARK: Properties and methods
+    
+    /// An online image service that features Charlotte, North Carolina's downtown area.
+    let imageServiceURL = URL(string: "https://sampleserver6.arcgisonline.com/arcgis/rest/services/CharlotteLAS/ImageServer")!
+    /// A list of rendering rule info supported by the service.
+    var rasterRenderingRules = [AGSRenderingRuleInfo]()
+    /// A dictionary to cache created raster layers.
+    var rasterLayers = [String: AGSRasterLayer]()
+    
+    /// Create a map.
+    ///
+    /// - Returns: An `AGSMap` object.
+    func makeMap() -> AGSMap {
+        let map = AGSMap(basemap: .streets())
+        // Create a raster layer from the raster and add it to the map.
+        let imageServiceRaster = AGSImageServiceRaster(url: imageServiceURL)
+        map.operationalLayers.add(makeRasterLayer(raster: imageServiceRaster))
+        // Load the raster and get a list of rendering rule info supported by the service.
+        imageServiceRaster.load { [weak self, unowned imageServiceRaster] error in
+            guard let self = self else { return }
+            if let serviceInfo = imageServiceRaster.serviceInfo, let extent = serviceInfo.fullExtent {
+                self.mapView.setViewpoint(AGSViewpoint(targetExtent: extent), completion: nil)
+                self.rasterRenderingRules = serviceInfo.renderingRuleInfos
+                self.chooseRenderingRuleBarButtonItem.isEnabled = true
+            } else if let error = error {
+                self.presentAlert(error: error)
+            }
+        }
+        return map
+    }
+    
+    /// Create a raster layer from an image service raster with an optional rendering rule.
+    ///
+    /// - Parameters:
+    ///   - raster: An `AGSImageServiceRaster` object.
+    ///   - renderingRule: An optional `AGSRenderingRule` object.
+    /// - Returns: An `AGSRasterLayer` object.
+    func makeRasterLayer(raster: AGSImageServiceRaster, renderingRule: AGSRenderingRule? = nil) -> AGSRasterLayer {
+        raster.renderingRule = renderingRule
+        let rasterLayer = AGSRasterLayer(raster: raster)
+        return rasterLayer
+    }
+    
+    // MARK: Actions
+    
+    @IBAction func chooseRenderingRule(_ sender: UIBarButtonItem) {
+        let alertController = UIAlertController(
+            title: "Choose a raster rendering rule to apply to the image service raster.",
+            message: nil,
+            preferredStyle: .actionSheet
+        )
+        rasterRenderingRules.forEach { ruleInfo in
+            let action = UIAlertAction(title: ruleInfo.name, style: .default) { _ in
+                let map = self.mapView.map!
+                // Clear all raster layers before adding new one.
+                map.operationalLayers.removeAllObjects()
+                let rasterLayer: AGSRasterLayer
+                if self.rasterLayers[ruleInfo.name] != nil {
```

Comment:
Use `if let existingLayer = self.rasterLayers[ruleInfo.name] { ... }` instead of subscripting twice.